### PR TITLE
prometheus: drop labels with duplicate nodename entries

### PR DESCRIPTION
**Pull request by ggilmore**
Follows https://github.com/sourcegraph/deploy-sourcegraph-dogfood-k8s/pull/5616, https://github.com/sourcegraph/deploy-sourcegraph-dogfood-k8s/pull/5613

This PR tweaks the Prometheus configuration to drop labels with an empty `nodename` parameter. 

Since https://github.com/sourcegraph/deploy-sourcegraph-helm/pull/194, there seems to be duplicate metrics (one with an empty `nodename`, one with a `populated` one). I'm unsure of the exact cause, but I can speculate that:
- perhaps the k8s api is returning empty responses in some cases
- perhaps the metric relabelling process ends up duplicating a time series 

This PR resolves this situtation by dropping the metrics with  empty `nodename` labels.

### Checklist

- [x] Follow the [manual testing process](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/TEST.md)
- [x] Update [changelog](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/charts/sourcegraph/CHANGELOG.md) (not needed - change hasn't released yet)

### Test plan

Running on k8s.sgdev.org, see the following commits:

https://github.com/sourcegraph/deploy-sourcegraph-dogfood-k8s/pull/5613
https://github.com/sourcegraph/deploy-sourcegraph-dogfood-k8s/pull/5616

### CODE REVIEW

**Comment by ggilmore** [comment]
cc @mucles When this gets merged, would you mind deploying this fix to the scaletesting cluster?

File: charts/sourcegraph/templates/prometheus/prometheus.ConfigMap.yaml [hunk 1]
```diff
@@ -249,7 +254,7 @@ data:
         target_label: ns
       # Sourcegraph specific customization. We want to add a label to every 
       # metric that indicates the node it came from.
-      - source_labels: [__meta_kubernetes_endpoint_node_name]
+      - source_labels: [__meta_kubernetes_pod_node_name]
```
Comment (keegancsmith) [inline charts/sourcegraph/templates/prometheus/prometheus.ConfigMap.yaml:257]: this seems unrelated? what was the reason for this change?

**Review by keegancsmith — APPROVED** [review]

File: charts/sourcegraph/templates/prometheus/prometheus.ConfigMap.yaml (hunk 1 above)
Comment (ggilmore) [reply charts/sourcegraph/templates/prometheus/prometheus.ConfigMap.yaml:257]: This was just a typo that existed in the original version of the prometheus node-exporter changes that I silently fixed here. `__meta_kubernetes_pod_node_name` is the correct name for this label (since we're scraping pods, not endpoints).  I figured this wasn't too bad since nobody has deployed this helm chart yet.
